### ISSUE TITLE
Implement time-varying input sets

### ISSUE THREAD

**Issue by mforets** [JuliaReach/Reachability.jl#19]
To handle x' = Ax + u(t), where u(t) belongs to a sequence of time-varying sets U(k) for k = 0, 1, ..., N.

**Comment by schillic** [comment]
I noticed that we might not even properly support this type in the input.
It currently only makes sense if `δ` is fixed by the user.
Otherwise the discretization that is used in this type can diverge from the `δ` that we choose.

We should maybe change this to a map from time moments to sets.
This can be implemented with an array of pairs `[(t0=0, s0), (t1, s1), ..., (tn, sn)]` and is interpreted as using `s0` in the interval [0, `t1`], then using `s1` until `t2`, and so on.

**Comment by mforets** [comment]
yes, assuming fixed step size delta. can we do the general case in a later stage?

**Comment by schillic** [comment]
Okay for later, but then I suggest that we should currently crash if the time steps do not coincide.

**Comment by schillic** [comment]
An idea could be that we add a variant where we compute the inputs lazily. This is interesting for the check mode with early termination.
